apps/web/src/hooks/useSubgraphHealth.ts

```diff
@@ -113,56 +124,66 @@ const useSubgraphHealth = ({ chainId, subgraph }: { chainId: ChainId; subgraph?:
               }
             }
           `,
-            ),
-            currentBlockNumber
-              ? Promise.resolve(currentBlockNumber)
-              : publicClient({ chainId })
-                  ?.getBlockNumber()
-                  .then((blockNumber) => Number(blockNumber)),
-          ])
+          ),
+          currentBlockNumber ? Promise.resolve(currentBlockNumber) : Promise.resolve(initialBlock),
+        ])
 
-          const [indexingStatusForCurrentVersion] = indexingStatuses
+        const [indexingStatusForCurrentVersion] = indexingStatuses
 
-          const isHealthy = indexingStatusForCurrentVersion?.health === 'healthy'
-          const chainHeadBlock = parseInt(indexingStatusForCurrentVersion?.chains[0]?.chainHeadBlock.number)
-          const latestBlock = parseInt(indexingStatusForCurrentVersion?.chains[0]?.latestBlock.number)
-          const blockDifference = currentBlock - latestBlock
-          // Sometimes subgraph might report old block as chainHeadBlock, so its important to compare
-          // it with block retrieved from simpleRpcProvider.getBlockNumber()
-          const chainHeadBlockDifference = currentBlock - chainHeadBlock
-          if (
-            !isHealthy ||
-            blockDifference > NOT_OK_BLOCK_DIFFERENCE ||
-            chainHeadBlockDifference > NOT_OK_BLOCK_DIFFERENCE
-          ) {
-            setSgHealth({ status: SubgraphStatus.NOT_OK, currentBlock, chainHeadBlock, latestBlock, blockDifference })
-          } else if (
-            blockDifference > WARNING_BLOCK_DIFFERENCE ||
-            chainHeadBlockDifference > WARNING_BLOCK_DIFFERENCE
-          ) {
-            setSgHealth({ status: SubgraphStatus.WARNING, currentBlock, chainHeadBlock, latestBlock, blockDifference })
-          } else {
-            setSgHealth({ status: SubgraphStatus.OK, currentBlock, chainHeadBlock, latestBlock, blockDifference })
+        const isHealthy = indexingStatusForCurrentVersion?.health === 'healthy'
+        const chainHeadBlock = parseInt(indexingStatusForCurrentVersion?.chains[0]?.chainHeadBlock.number)
+        const latestBlock = parseInt(indexingStatusForCurrentVersion?.chains[0]?.latestBlock.number)
+        const blockDifference = currentBlock - latestBlock
+        // Sometimes subgraph might report old block as chainHeadBlock, so it's important to compare
+        // it with block retrieved from simpleRpcProvider.getBlockNumber()
+        const chainHeadBlockDifference = currentBlock - chainHeadBlock
```

Comment:
sometimes explorer might faster than onchain block (due to slow interval), make sure it won't show negative delay on UI